File: 1231-replace-elements-with-greatest-element-on-right-side/replace-elements-with-greatest-element-on-right-side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        n = len(arr)
        maxi = -1
        for i in range(n-1, -1, -1):
            temp = arr[i]
            arr[i] = maxi
            maxi = max(temp, maxi)

        return arr
            

File: 1231-replace-elements-with-greatest-element-on-right-side/replace-elements-with-greatest-element-on-right-side.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int n = arr.length;
        int maxi = -1;
        for (int i = n-1; i > -1; i--){
            int temp = arr[i];
            arr[i] = maxi;
            maxi = Math.max(temp, maxi);
        }

        return arr;
    }
}